app/components/common/SearchableAddress.js

```diff
@@ -58,33 +58,55 @@ class SearchableAddress extends Component {
 
   handleSearchChange = (e, { value }) => {
     this.setState({ value })
+    this.filterSearchInput()
+  }
+
+  filterSearchInput = () => {
+    const value = this.state.value
+
+    // if (this.state.value.length < 1) return this.setState(initialState)
+    if (!tfchain.wallet_address_is_valid(value)) {
```

Comment:
if you do this you can avoid branching on an `if else` branch by directly assigning the state as follows:

```javascript
this.setState({ addressError: !tfchain.wallet_address_is_valid(value)
```